crates/sui-indexer-alt-e2e-tests/tests/consistent_store_list_objects_by_type_tests.rs

```diff
@@ -0,0 +1,370 @@
+    assert_eq!(
+        list_objects_by_type(&cluster, "0x2::bag", None, Some(50))
+            .await
+            .unwrap(),
+        (bags.values().map(repr).collect(), None)
+    );
+
+    // All tables, by fully-qualified name filter
+    assert_eq!(
+        list_objects_by_type(&cluster, "0x2::table::Table", None, Some(50))
+            .await
+            .unwrap(),
+        (
+            tu8s.values().chain(tu64s.values()).map(repr).collect(),
+            None
+        )
+    );
+
+    // All Table<u64, u64>s
+    assert_eq!(
+        list_objects_by_type(&cluster, "0x2::table::Table<u64, u64>", None, Some(50))
+            .await
+            .unwrap(),
+        (tu64s.values().map(repr).collect(), None)
+    );
+
+    // Try to paginate Table<u8, u64>s -- none should exist.
+    assert_eq!(
+        list_objects_by_type(&cluster, "0x2::table::Table<u8, u64>", None, Some(50))
+            .await
+            .unwrap(),
+        (vec![], None)
+    );
+
+    // Test pagination
+    let mut after = None;
+    let mut results = vec![];
+    loop {
+        let (page, next) = list_objects_by_type(&cluster, "0x2::table", after.clone(), Some(5))
+            .await
+            .unwrap();
+
+        results.extend(page);
+        after = next;
+        if after.is_none() {
+            break;
+        }
+    }
+
+    // Pagination should return the same results as fetching all at once
+    assert_eq!(
+        results,
+        tu8s.values()
+            .chain(tu64s.values())
+            .map(repr)
+            .collect::<Vec<_>>()
+    );
+
+    // Pass in a bad type filter (malformed).
+    let err = list_objects_by_type(&cluster, "not::a::valid::type<>", None, Some(20))
+        .await
+        .unwrap_err();
+
+    assert_eq!(err.code(), tonic::Code::InvalidArgument);
+    assert_eq!(
+        err.message(),
+        "Bad 'object_type' filter, expected: package[::module[::name[<type, ...>]]]"
+    );
+
+    // Missing type filter
+    let err = list_objects_by_type(&cluster, "", None, Some(20))
+        .await
+        .unwrap_err();
+
+    assert_eq!(err.code(), tonic::Code::InvalidArgument);
+    assert_eq!(err.message(), "Missing 'object_type' filter");
+}
+
+fn repr((i, v, d): &ObjectRef) -> (String, u64, String) {
+    (
+        i.to_canonical_string(/* with_prefix */ true),
+        v.value(),
+        d.base58_encode(),
+    )
+}
+
+/// Run a transaction on `cluster` signed by a fresh funded account that creates a `Bag` with
+/// `size` many elements, owned by `kind`. The purpose of this is to create an object that isn't a
+/// coin. `ty` can be any numeric Move type.
+fn create_bag(cluster: &mut FullCluster, kind: OwnerKind, ty: TypeTag, size: u64) -> ObjectRef {
+    let (sender, kp, gas) = cluster
+        .funded_account(DEFAULT_GAS_BUDGET)
+        .expect("Failed to fund account");
+
+    let mut builder = ProgrammableTransactionBuilder::new();
+
+    let bag = builder.programmable_move_call(
```

Comment:
Imagine how much easier this will become when we have transactional test support..

Reply:
FWIW, I would probably do transactional testing via the GraphQL integration.